whittle/models/gpt/blocks/causal_self_attention.py

```diff
@@ -50,27 +50,8 @@ def set_sub_network(
     ):
         self.sub_network_n_embd = sub_network_n_embd
         self.sub_network_n_head = sub_network_n_head
-        if sub_network_query_groups is None:
-            if self.config.n_query_groups == 1:
-                self.sub_network_query_groups = 1
-            elif self.sub_network_n_head % self.config.n_query_groups == 0:
-                self.sub_network_query_groups = self.config.n_query_groups
-            else:
-                self.sub_network_query_groups = self.sub_network_n_head // (
-                    self.config.n_head // self.config.n_query_groups
-                )
-        else:
-            self.sub_network_query_groups = sub_network_query_groups
-        if self.config.fix_head_size:
-            if sub_network_head_size is None:
-                self.sub_network_head_size = self.config.head_size
-            else:
-                self.sub_network_head_size = sub_network_head_size
-
-        else:
-            self.sub_network_head_size = (
-                self.sub_network_n_embd // self.sub_network_n_head
-            )
+        self.sub_network_query_groups = sub_network_query_groups
+        self.sub_network_head_size = sub_network_head_size
```

Comment:
The way this is initialized currently, they can never be none

Reply:
Sorry, I meant to tag lines 48-49

```
      sub_network_query_groups=None,
      sub_network_head_size=None,
```

Since now they cannot be None, we might want to change it to

```
      sub_network_query_groups: int,
      sub_network_head_size: int,
```

Reply:
I think they should be either `None` or int no? Since they are `None` by default?
```
sub_network_query_groups: int | None,
sub_network_head_size: int | None,
```

Reply:
@aaronkl yes, but I think they should not be allowed to be `None` anymore - the computation is done in model.py and hence `None` is not a valid value here anymore